runtime/gc_structs/VMThreadStackSlotIterator.cpp

```diff
@@ -152,3 +152,43 @@ GC_VMThreadStackSlotIterator::scanSlots(
 
 	VM_VMHelpers::walkContinuationStackFramesWrapper(vmThread, continuationObjectPtr, &stackWalkState);
 }
+
+#if JAVA_SPEC_VERSION >= 19
+void
+GC_VMThreadStackSlotIterator::scanSlots(
+			J9VMThread *vmThread,
+			J9VMContinuation *continuation,
+			void *userData,
+			J9MODRON_OSLOTITERATOR *oSlotIterator,
+			bool includeStackFrameClassReferences,
+			bool trackVisibleFrameDepth
+		)
+{
+	J9StackWalkState stackWalkState;
+	J9JavaVM *vm = vmThread->javaVM;
+
+	stackWalkState.objectSlotWalkFunction = vmThreadStackDoOSlotIterator;
+	stackWalkState.userData1 = (void *)oSlotIterator;
+	stackWalkState.userData2 = (void *)vm;
+	stackWalkState.userData3 = userData;
+
+	stackWalkState.flags = J9_STACKWALK_ITERATE_O_SLOTS | J9_STACKWALK_DO_NOT_SNIFF_AND_WHACK;
+
+	if (trackVisibleFrameDepth) {
+		stackWalkState.skipCount = 0;
+		stackWalkState.flags |= J9_STACKWALK_VISIBLE_ONLY;
+	} else {
+		if (NULL != vm->collectJitPrivateThreadData) {
+			stackWalkState.frameWalkFunction = vmThreadStackFrameIterator;
+			stackWalkState.flags |= J9_STACKWALK_ITERATE_FRAMES;
+		}
+		stackWalkState.flags |= J9_STACKWALK_SKIP_INLINES;
+	}
+
+	if (includeStackFrameClassReferences) {
+		stackWalkState.flags |= J9_STACKWALK_ITERATE_METHOD_CLASS_SLOTS;
+	}
```

Comment:
could you introduce a local method like initializeStackWalkState(ForGC?) and share this pattern with a couple of 2 other scanSlots methods in this file?